tests/unit_tests/dft/include/test_common.hpp

```diff
@@ -0,0 +1,228 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_TEST_COMMON_HPP
+#define ONEMKL_TEST_COMMON_HPP
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+template <typename T>
+struct complex_info {
+    using real_type = T;
+    static const bool is_complex = false;
+};
+
+template <typename T>
+struct complex_info<std::complex<T>> {
+    using real_type = T;
+    static const bool is_complex = true;
+};
+
+template <typename T>
+constexpr bool is_complex() {
+    return complex_info<T>::is_complex;
+}
+template <typename T>
+constexpr int num_components() {
+    return is_complex<T>() ? 2 : 1;
+}
+
+template <typename fp>
+typename std::enable_if<!std::is_integral<fp>::value, bool>::type check_equal(fp x, fp x_ref,
+                                                                              int error_mag) {
```

Comment:
```suggestion
bool check_equal(fp x, fp x_ref, int error_mag) {
  static_assert(!std::is_integral_v<fp>, "check_equal is only for floating point values and complex values");
```
I don't think enable_if is necessary here, a static_assert would be suitable. enable_if would be needed if you were going for some SFINAE.

Reply:
Done in [56d18e6](https://github.com/oneapi-src/oneMKL/pull/261/commits/56d18e60d45dc3a0b617eef414eb0e84cbe6d7c3).